### DISCUSSION TITLE
Remaining Capacity in kWh additional to Ah

### DISCUSSION THREAD

**Discussion by dominikfe** [syssi/esphome-jk-bms#131] [Ideas]
Would be a nice to have feature and should not be too much effort?

**Comment by syssi** [comment]
Could you try this proposal: https://github.com/syssi/esphome-jk-bms/issues/80

If it's working for you I would like to add it to the example configurations.

**Reply by syssi** [reply]
Oh, you are asking about the state of charge in kWh. Let me think about it.

**Reply by syssi** [reply]
To calculate the remaining kWh we need the nominal voltage of the battery. You should use a template sensor:

```
sensor:
  - platform: jk_bms_ble
    jk_bms_ble_id: bms0
    capacity_remaining:
      name: "${name} capacity remaining"
      id: capacity_remaining_sensor
      
  - platform: template
    name: "${name} energy remaining"
    unit_of_measurement: "kWh"
    device_class: "energy"
    lambda: |-
       auto val = id(capacity_remaining_sensor).state * 48.0f;
       if (isnan(val))
          return {};
       return val;
    update_interval: 10s

```
